cli/src/anker_state.rs

```diff
@@ -0,0 +1,126 @@
+use anchor_lang::prelude::Pubkey;
+use anker::{
+    find_instance_address, find_reserve_authority, find_st_sol_reserve_account,
+    find_ust_reserve_account,
+    state::Anker,
+    token::{BLamports, MicroUst},
+};
+use lido::{state::Lido, token::StLamports};
+use solana_program::{instruction::Instruction, program_pack::Pack};
+use solana_sdk::account::ReadableAccount;
+
+use crate::{snapshot::Result, SnapshotConfig};
+
+#[derive(Default)]
+pub struct AnkerState {
+    pub anker: Anker,
+    pub anker_program_id: Pubkey,
+
+    pub b_sol_total_supply_amount: BLamports,
+    pub pool_st_sol_account: Pubkey,
+    pub pool_ust_account: Pubkey,
+    pub ust_mint: Pubkey,
+    pub pool_mint: Pubkey,
+    pub pool_fee_account: Pubkey,
+    pub ust_reserve_balance: MicroUst,
+    pub st_sol_reserve_balance: StLamports,
+}
+
+impl AnkerState {
+    pub fn new(
+        config: &mut SnapshotConfig,
+        anker_program_id: Pubkey,
+        solido_address: &Pubkey,
+        solido: &Lido,
+    ) -> Result<Self> {
+        let (anker_instance, _anker_bump_seed) =
+            find_instance_address(&anker_program_id, solido_address);
+        let anker = config.client.get_anker(&anker_instance)?;
+
+        let token_swap_account = config.client.get_account(&anker.token_swap_pool)?;
+        let token_swap = spl_token_swap::state::SwapV1::unpack(token_swap_account.data())?;
+
+        let (anker_ust_reserve, _anker_ust_reserve_bump_seed) =
+            find_ust_reserve_account(&anker_program_id, solido_address);
+        let ust_reserve_balance =
+            MicroUst(config.client.get_spl_token_balance(&anker_ust_reserve)?);
+
+        let (anker_st_sol_reserve, _anker_st_sol_reserve_bump_seed) =
+            find_st_sol_reserve_account(&anker_program_id, solido_address);
+        let st_sol_reserve_balance =
+            StLamports(config.client.get_spl_token_balance(&anker_st_sol_reserve)?);
+
+        let b_sol_mint_account = config.client.get_spl_token_mint(&anker.b_sol_mint)?;
+        let b_sol_total_supply_amount = BLamports(b_sol_mint_account.supply);
+
+        let (ust_account, ust_mint, st_sol_account) =
```

Comment:
Maybe call these `swap_ust_account` and `swap_st_sol_account`, to avoid confusing them with the reserve?